src/cloud/auth.rs

```diff
@@ -27,16 +28,48 @@ struct UserSession {
     auth_url: String,
 }
 
+#[derive(Debug, serde::Deserialize)]
+struct User {
```

Comment:
You can either do `allow(dead_code)` for the struct to suppress warnings, if you want to validate structure. Or remove unused fields, for forward compatibility, serde_json allows extra fields in JSON by default. Or combination of both.